Run operator within "ads opctl run"

mrDzurb opened the pull request:
## Description
https://jira.oci.oraclecorp.com/browse/ODSC-48941

* Modifies the operator's execution to be compatible with the `ads opctl run` command. This modification enables operators to be executed like other workloads within the `ads opctl run` command.
**Example:**
```
ads opctl run -f operator.yaml -b jobs
ads opctl run -f operator.yaml -b local
ads opctl run -f operator.yaml -b dataFlow
```
The all configurations will be auto prepared based on the ads config and environment variables.
However the `ads opctl run` will not support the all features supported by `ads operator run`. 

* Adds `BackendFactory` class which contains the factory method to create the operator's compatible backend objects. 
* Adds unit tests

![image](https://github.com/oracle/accelerated-data-science/assets/5256765/7550d0bc-1436-4259-bbe2-fc2eebb23b47)

ahosler reviewed (approved):
Awesome work!
2 questions:

1. I can run `ads operator run -f forecast.yaml` or `ads optcl run -f forecast.yaml -b local`, but not `ads optcl run -f forecast.yaml`. Can we have a default backend?
2. Should we make any changes to the docs? Should we encourage users to use "opctl" over "operator"?
Neither question is critical to merging, PR looks great!

mrDzurb commented:
> Awesome work! 2 questions:
> 
> 1. I can run `ads operator run -f forecast.yaml` or `ads optcl run -f forecast.yaml -b local`, but not `ads optcl run -f forecast.yaml`. Can we have a default backend?
> 2. Should we make any changes to the docs? Should we encourage users to use "opctl" over "operator"?
> 
> Neither question is critical to merging, PR looks great!

Thanks Allen, I've addressed your comments and updated documentation. In the documentation i put some explanation the difference between running operator with ```ads opctl run``` and ```ads operator run```. 

I also added a few new features. 

```
ads operator init -t forecast --merge-config
```

The `--merge-config` allows to merge operator's specification with the backend runtime. Basically all config in one file. This will allow to run operators like we run regular jobs. 

```
ads opctl run -f operator.yaml
```
